plugins/ui/edgy.nix

```diff
@@ -0,0 +1,323 @@
+          })
+        ) [ ] "List of the ${name} edgebar configurations.";
+    in
+    {
+      left = mkViewOptsOption "left";
+      bottom = mkViewOptsOption "bottom";
+      right = mkViewOptsOption "right";
+      top = mkViewOptsOption "top";
+
+      options =
+        mapAttrs
+          (_: defaultSize: {
+            size = helpers.defaultNullOpts.mkUnsignedInt defaultSize ''
+              Size of the short edge of the edgebar.
+              For edgebars, this is the minimum width.
+              For panels, minimum height.
+            '';
+
+            wo = helpers.mkNullOrOption (with types; attrsOf anything) ''
+              View-specific window options.
+            '';
+          })
+          {
+            left = 30;
+            bottom = 10;
+            right = 30;
+            top = 10;
+          };
+
+      animate = {
+        enabled = helpers.defaultNullOpts.mkBool true ''
+          Whether to enable animations.
+        '';
+
+        fps = helpers.defaultNullOpts.mkUnsignedInt 100 ''
+          Frames per second.
+        '';
+
+        cps = helpers.defaultNullOpts.mkUnsignedInt 120 ''
+          Cells per second.
+        '';
+
+        on_begin = helpers.defaultNullOpts.mkLuaFn ''
+          function()
+            vim.g.minianimate_disable = true
+          end
+        '' "Callback for the beginning of animations.";
+
+        on_end = helpers.defaultNullOpts.mkLuaFn ''
+          function()
+            vim.g.minianimate_disable = false
+          end
+        '' "Callback for the ending of animations.";
+
+        # This option accepts an attrs or a lua string.
+        # Hence, we use `mkOption` to convert the string to raw lua in `apply`.
+        spinner = mkOption {
+          type =
+            with helpers.nixvimTypes;
+            nullOr (
+              either strLua (submodule {
+                freeformType = attrsOf anything;
+                options = {
+                  frames = helpers.defaultNullOpts.mkListOf types.str [
+                    "⠋"
+                    "⠙"
+                    "⠹"
+                    "⠸"
+                    "⠼"
+                    "⠴"
+                    "⠦"
+                    "⠧"
+                    "⠇"
+                    "⠏"
+                  ] "Frame characters.";
+
+                  interval = helpers.defaultNullOpts.mkUnsignedInt 80 ''
+                    Interval time between two consecutive frames.
+                  '';
+                };
+              })
+            );
+          default = null;
+          example = "require('noice.util.spinners').spinners.circleFull";
+          apply = v: if isString v then helpers.mkRaw v else v;
+          description = helpers.defaultNullOpts.mkDesc {
+            frames = [
+              "⠋"
+              "⠙"
+              "⠹"
+              "⠸"
+              "⠼"
+              "⠴"
+              "⠦"
+              "⠧"
+              "⠇"
+              "⠏"
+            ];
```

Comment:
I'd be tempted to put the `frames` default in the let-block, but as it's only used twice this is fine if preferred.

Reply:
That's a good idea. Done.